cloud/pkg/admissioncontroller/mutate_offlinemigration.go

```diff
@@ -49,28 +43,28 @@ func mutateOfflineMigration(review admissionv1beta1.AdmissionReview) *admissionv
 }
 
 func generatePatch(tolerations []corev1.Toleration) []patchMapValue {
-	patch := []patchMapValue{{
-		Op:   "add",
-		Path: "/spec/template/spec/tolerations",
-		Value: []map[string]string{{
-			"key":      corev1.TaintNodeUnreachable,
-			"operator": Exists,
-			"effect":   NoExecute,
-		}},
-	}}
-	if len(tolerations) > 0 {
-		for _, toleration := range tolerations {
-			if toleration.Key == corev1.TaintNodeUnreachable {
-				if toleration.Effect == "NoExecute" &&
-					toleration.Operator == "Exists" && toleration.TolerationSeconds == nil {
-					return nil
-				}
-				toleration.TolerationSeconds = nil
-				patch[0].Op = "replace"
-			}
+	var hasUnreachable bool
+	currentTolerations := make([]interface{}, 0, len(tolerations)+1)
+	for _, v := range tolerations {
+		if v.Key == corev1.TaintNodeUnreachable {
```

Comment:
```suggestion
		if v.Key == corev1.TaintNodeUnreachable && v.Effect == corev1.TaintEffectNoExecute  {
```
How about add `&& v.Effect == corev1.TaintEffectNoExecute`, if v.Effect is noschedule, we still need to add the `NoExecute`, which is related to the eviction.

Reply:
`NoSchedule `: Kubernetes will not schedule the pod onto that node.
`NoExecute `: the pod will be evicted from the node (if it is already running on the node), and will not be scheduled onto the node (if it is not yet running on the node).

I think `NoSchedule` will be used in the future, and it should not be forcibly overwritten here.

Reply:
If we only campare the `Key`, and if the effect is `NoSchedule `, we still will set hasUnreachable to true, and will not add the toleration with effect `NoExecute`? And if pod tolerate the effect `NoExecute`, it will not be evicted from node. Do I miss anything? ;-)

Reply:
No, what you understand is correct.
Some design issues are involved here, and we can have some discussion.

Reply:
Okey, let's have another discussion next Monday :)